cucumber-core/src/main/java/io/cucumber/core/plugin/TeamCityPlugin.java

```diff
@@ -101,6 +104,11 @@ public class TeamCityPlugin implements EventListener {
     private static final Pattern ANNOTATION_GLUE_CODE_LOCATION_PATTERN = Pattern.compile("^(.*)\\.(.*)\\([^:]*\\)");
     private static final Pattern LAMBDA_GLUE_CODE_LOCATION_PATTERN = Pattern.compile("^(.*)\\.(.*)\\(.*:.*\\)");
 
+    private static final Pattern[] COMPARE_PATTERNS = new Pattern[] {
+            Pattern.compile("^expected:\\s*(.*)\\s+but was:\\s*(.*)",
```

Comment:
Looking at the `AssertionFailureBuilder` from JUnit 5 there may be a reason and message that are prefixed to the different values. 

https://github.com/junit-team/junit5/blob/main/junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertionFailureBuilder.java#L144
https://github.com/junit-team/junit5/blob/main/junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertionFailureBuilder.java#L148

To keep the test fidelity high it might be good to use the `AssertionFailureBuilder` to create the exception messages instead.